bundles/framework/layerselector2/view/FilterButtons.js

```diff
@@ -0,0 +1,175 @@
+        this.filterTemplate = jQuery('<div class="filter filter-border">'+
+                                            '<center><div class="filter-icon">'+
+                                            '</div><div class="filter-text"></div></center>'+
+                                        '</div>');
+        this.filterButtons = [];
+
+        Oskari.makeObservable(this);
+    }, {
+        create: function (tabs, toolText, tooltip, iconClassActive, iconClassDeactive, filterName) {
+            this.layerTabs = tabs;
+            var me = this;
+            var loc = Oskari.getLocalization('LayerSelector').layerFilter;
+
+            if ( me.filterIsCreated(filterName) ) {
+                me.render();
+                return;
+            } else {
+                var filterButton = me.filterTemplate.clone()
+
+                filterButton.attr('data-filter', filterName);
+                filterButton.find('.filter-text').html(toolText);
+                filterButton.attr('title', tooltip);
+                filterButton.find('.filter-icon').addClass('filter-' + filterName);
+                filterButton.find('.filter-icon').addClass(iconClassDeactive);
+               
+                me.filterButtons.push({
+                    name: filterName,
+                    element: filterButton
+                });
+
+                filterButton.unbind('click');
+                filterButton.bind('click', function(evt) {
+                    var filterIcon = jQuery(evt.target).parent().find('.filter-icon.' + 'filter-' + filterName);
+                    me.deactivateAllFilters(filterName);
+                    if (filterIcon.hasClass(iconClassDeactive)) {
+                        // Activate this filter
+                        me._setFilterIconClasses(filterName);
+                        me.activateFilter(filterName);
+                        me._setFilterTooltip(filterName, loc.tooltips.remove);
+                    } else {
+                        // Deactivate all filters
+                        me.deactivateAllFilters();
+                    }
+                });
+                me.render();
+            }
+        },
+        filterIsCreated: function (filterName) {
+            var created = false;
+            this.filterButtons.filter(function(button) {
+                if ( button.name === filterName ) {
+                    created = true;
+                }
+            });
+            return created;
+        },
+          /**
+         * Set filter button tooltip
+         * @method  @private _setFilterTooltip
+         * @param {String} filterName filter name
+         * @param {String} tooltip    tooltip
+         */
+        _setFilterTooltip: function(filterName, tooltip) {
+            var me = this;
+            me.layerTabs.forEach(function(tab) {
+                var filterContainer = tab.getTabPanel().getContainer().find('.layerselector2-layer-filter');
```

Comment:
You should pass the element for the buttons in constructor instead of this component needing very detailed information about the structure of the parent component.